tests/integ/conftest.py

```diff
@@ -209,6 +209,8 @@ def session(
     session = (
         Session.builder.configs(db_parameters)
         .config("local_testing", local_testing_mode)
+        # Enable thread safe session for testing
+        .config("PYTHON_SNOWPARK_ENABLE_THREAD_SAFE_SESSION", True)
```

Comment:
for tests, we are adding multithreading gates which will run all snowpark tests in enabled and disabled mode: https://github.com/snowflakedb/snowpark-python/pull/2336

Reply:
we should do it like how others are turning the feature on by flipping on the feature flag in our testing account, so that the change can be turned on and off without pr changes

Reply:
adding that in the PR mentioned above.